patches/await-semaphore+0.1.3.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/node_modules/await-semaphore/index.ts b/node_modules/await-semaphore/index.ts
```

Comment:
The `await-semaphore` package contains TypeScript files in its distributed form. This is not necessary or recommended for TypeScript-compatible libraries (rather, it is enough to publish type _definition_ files). The issue for this is logged here: https://github.com/notenoughneon/await-semaphore/issues/6. In the meantime, as our code works fine presently without any TypeScript files, we do not need these in order to use this library. This makes the new `tsc` step in `yarn lint` pass.

Reply:
So, to make sure I understand correctly, this code is not used by the extension, but we need this patch to get lint passing?

Reply:
We do use this package (it looks like [at the top of MetamaskController](https://github.com/MetaMask/metamask-extension/blob/b8fc1ae671c153d84de6caa254de3e55c153de75/app/scripts/metamask-controller.js#L13)). However, the way this package has been bundled and published to NPM is incorrect as it includes TypeScript files.

Reply:
It looks like we're removing the Semaphore class that Mutex extends in this patch. Does this not break when `this.createVaultMutex.acquire();` runs in the MetamaskController?

Reply:
So, we aren't removing all the code in `await-semaphore`. You can see [here](https://unpkg.com/browse/await-semaphore@0.1.3/) that the version of this package published to NPM includes both JavaScript and TypeScript files. We are just removing the TypeScript files, but we are still using the JavaScript files. So `this.createVaultMutex.acquire();` should still work as before.